virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -239,16 +233,18 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
 
     return windowId;
   }
-
+   
+ 
   getWindow(plugin: MVDHosting.DesktopPluginDefinition): MVDWindowManagement.WindowId | null {
-    const desktopWindows = this.pluginMap.get(plugin.getIdentifier());
-    if (desktopWindows !== undefined) {
-      return desktopWindows[0].windowId;
+    const applicationWindowID = this.applicationManager.getAppWindowID(plugin.getIdentifier());
```

Comment:
Why did you move windowID decision making outside of windowmanager service and into applicationmanager? window manager manages windows, application manager manages applications. applications are spawned into viewports, which the viewport manager handles. viewports go into windows, which the window manager handles.
Some of the existing code is wrong, but I think this would make it more wrong. I'd rather move all window logic from application manager into window manager. This will make more sense when Suneeth finishes multi-window support per application.

Reply:
I thought either app manager or window manager should be more powerful on something to make it possible to launch multiple instances of one plugin. So I moved some of window manager part into app manager, which made window manager simpler and app manager more complicated. Sure I could do verse vice.